### PR TITLE
Makefile: add new recipes and linter features

### PR DESCRIPTION
**Pull request by thrasher-**
# Description

Introduces a new makefile with various recipes.

## Type of change

- [x] New feature (non-breaking change which adds functionality)

# How Has This Been Tested?

Travis and individual recipe testing.

# Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally and on Travis with my changes
- [x] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

File: exchanges/bitmex/bitmex_parameters.go [hunk 1]
```diff
@@ -110,10 +110,7 @@ func (p APIKeyParams) ToURLVals(path string) (string, error) {
 
 // IsNil checks to see if any values has been set for the paramater
 func (p APIKeyParams) IsNil() bool {
-	if p == (APIKeyParams{}) {
-		return true
-	}
-	return false
+	return (p == (APIKeyParams{}))
```
Comment (shazbert) [inline exchanges/bitmex/bitmex_parameters.go]: Drop outside brackets

File: exchanges/bitmex/bitmex_parameters.go [hunk 2]
```diff
@@ -148,10 +145,7 @@ func (p ChatGetParams) ToURLVals(path string) (string, error) {
 
 // IsNil checks to see if any values has been set for the paramater
 func (p ChatGetParams) IsNil() bool {
-	if p == (ChatGetParams{}) {
-		return true
-	}
-	return false
+	return (p == (ChatGetParams{}))
```
Comment (shazbert) [inline exchanges/bitmex/bitmex_parameters.go]: Drop outside brackets

File: exchanges/bitmex/bitmex_parameters.go [hunk 3]
```diff
@@ -179,10 +173,7 @@ func (p ChatSendParams) ToURLVals(path string) (string, error) {
 
 // IsNil checks to see if any values has been set for the paramater
 func (p ChatSendParams) IsNil() bool {
-	if p == (ChatSendParams{}) {
-		return true
-	}
-	return false
+	return (p == (ChatSendParams{}))
```
Comment (shazbert) [inline exchanges/bitmex/bitmex_parameters.go]: Drop outside brackets

File: exchanges/bitmex/bitmex_parameters.go [hunk 4]
```diff
@@ -239,10 +230,7 @@ func (p GenericRequestParams) ToURLVals(path string) (string, error) {
 
 // IsNil checks to see if any values has been set for the paramater
 func (p GenericRequestParams) IsNil() bool {
-	if p == (GenericRequestParams{}) {
-		return true
-	}
-	return false
+	return (p == (GenericRequestParams{}))
```
Comment (shazbert) [inline exchanges/bitmex/bitmex_parameters.go]: Drop outside brackets

File: exchanges/bitmex/bitmex_parameters.go [hunk 5]
```diff
@@ -1116,8 +1035,5 @@ func (p UserRequestWithdrawalParams) ToURLVals(path string) (string, error) {
 
 // IsNil checks to see if any values has been set for the paramater
 func (p UserRequestWithdrawalParams) IsNil() bool {
-	if p == (UserRequestWithdrawalParams{}) {
-		return true
-	}
-	return false
+	return (p == (UserRequestWithdrawalParams{}))
```
Comment (shazbert) [inline exchanges/bitmex/bitmex_parameters.go]: Drop outside brackets

File: portfolio/portfolio.go [hunk 6]
```diff
@@ -198,10 +198,7 @@ func (p *Base) UpdatePortfolio(addresses []string, coinType string) bool {
 			}
 			p.AddAddress(addresses[x], coinType, PortfolioAddressPersonal, result.ETH.Balance)
 		}
-		if errors > 0 {
-			return false
-		}
-		return true
+		return (errors == 0)
```
Comment (shazbert) [inline portfolio/portfolio.go]: Drop outside brackets

File: portfolio/portfolio.go [hunk 7]
```diff
@@ -261,16 +258,16 @@ func (p *Base) GetPersonalPortfolio() map[string]float64 {
 // getPercentage returns the percentage of the target coin amount against the
 // total coin amount.
 func getPercentage(input map[string]float64, target string, totals map[string]float64) float64 {
```
Comment (shazbert) [inline portfolio/portfolio.go]: RM `map[string]float64` type dec. for input var.

Comment (thrasher-) [reply portfolio/portfolio.go]: Gonna leave this one as is since the target is of type string

**Comment by thrasher-** [comment]
@shazbert all nitterinos resolved 👍

File: portfolio/portfolio.go (hunk 7 above)
Comment (gloriousCode) [reply portfolio/portfolio.go]: How about
`func getPercentage(target, totals map[string]float64, target string) float64` 
?

**Comment by shazbert** [comment]
utACK WELL DONE SIR!

File: portfolio/portfolio.go (hunk 7 above)
Comment (thrasher-) [reply portfolio/portfolio.go]: Out of scope for this PR, there's probabably a lot of others like this but is insignificant. Malalignment checks (mainly for structs) can be done in the future.

**Pull request merged**